### ISSUE TITLE
[EKS] [bug]: ARGO jobs on EKS Windows Nodes

### ISSUE THREAD

**Issue by JasonChinsen** [aws/containers-roadmap#245]
**Tell us about your request**
Running [Argo](https://github.com/argoproj/argo) jobs on Windows nodes

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**

When trying to run ARGO jobs on Windows nodes I get the following error:

```MountVolume.SetUp failed for volume "docker-sock" : hostPath type check failed: /var/run/docker.sock is not a socket file```

There is a bug report in the argoproj [826](https://github.com/argoproj/argo/issues/826) that looks like it was fixed in the Kubernetes project for Azure 

**Are you currently working around this issue?**
N/A

**Additional context**

This is part of the describe pod from the Argo job:

```
...
Volumes:
  podmetadata:
    Type:  DownwardAPI (a volume populated by information about the pod)
    Items:
      metadata.annotations -> annotations
  docker-lib:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/docker
    HostPathType:  Directory
  docker-sock:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/docker.sock
    HostPathType:  Socket
  default-token-swr6z:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-swr6z
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  beta.kubernetes.io/os=windows
...
```

**Attachments**

**Comment by kernelcoredump** [comment]
In Windows, the Docker API is typically accessed through a named pipe "npipe:////./pipe/docker_engine" that can be volume-mounted from host to container, but may only be fixed in a later version (1.14?) of k8s https://github.com/kubernetes/kubernetes/issues/67147

**Comment by JasonChinsen** [comment]
> In Windows, the Docker API is typically accessed through a named pipe "npipe:////./pipe/docker_engine" that can be volume-mounted from host to container, but may only be fixed in a later version (1.14?) of k8s [kubernetes/kubernetes#67147](https://github.com/kubernetes/kubernetes/issues/67147)

@kernelcoredump Thanks for pointing this out, good find!. It looks like that ticket ref's [69484](https://github.com/kubernetes/kubernetes/pull/69484) ... It is included in 1.13 https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md (I did a quick find in the changelogs since 1.11). Now we wait for 1.13.x EKS :D and windows AMI's
